src/Compiler/AbstractIL/ilx.fs

```diff
@@ -58,10 +72,15 @@ type IlxUnionSpec =
         let (IlxUnionSpec (IlxUnionRef (_, _, _, np, _), _)) = x in np
 
     member x.HasHelpers = let (IlxUnionSpec (IlxUnionRef (_, _, _, _, b), _)) = x in b
+
     member x.Alternatives = Array.toList x.AlternativesArray
+
     member x.Alternative idx = x.AlternativesArray[idx]
+
     member x.FieldDef idx fidx = x.Alternative(idx).FieldDef(fidx)
 
+    override x.ToString() = x.TypeRef.Name
```

Comment:
Can `TypeRef` here be null for some reason?

Reply:
Here it cannot, it is always filled.